src/helm/benchmark/adaptation/adapters/chat_adapter.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.scenarios.scenario import Instance
+from helm.common.request import Request
+from helm.benchmark.adaptation.adapters.in_context_learning_adapter import InContextLearningAdapter
+
+
+class ChatAdapter(InContextLearningAdapter):
+    """
+    Each `Instance` in a `Scenario` has a history of the format:
+
+    [
+        {"role": "user", "content": <user-content>},
+        {"role": "assistant", "content": <assistant-content>},
+        {"role": "user", "content": <user-content>},
+        ...
+    ]
+
+    """
+
+    def generate_requests(
+        self, eval_instance: Instance, train_trial_index: int, training_instances: List[Instance]
+    ) -> List[RequestState]:
+        assert eval_instance.extra_data
+        messages = [
+            {"role": message["role"], "content": message["content"]}
+            for message in eval_instance.extra_data["conversation"]
+        ]
```

Comment:
Can we add `instance.input.messages` and use that instead of `extra_data` before we launch? OK to do in a separate PR.

Reply:
Wait, this is for IFEval, not for OmniMath?